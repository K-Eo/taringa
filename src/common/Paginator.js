import React from 'react'

const Paginator = ({
  currentPage,
  isLastPage,
  isLoading,
  onClickNext,
  onClickPrevious,
}) => {
  if (isLoading) {
    return null
  }

  return (
    <div className="Paginator">
      {currentPage > 1 && (
        <button
          className="btn btn-primary btn-sm Paginator-previous"
          onClick={onClickPrevious}
        >
          Anterior
        </button>
      )}
      &nbsp;
      {!isLastPage && (
        <button
          className="btn btn-primary btn-sm Paginator-next"
          onClick={onClickNext}
        >
          Siguiente
        </button>
      )}
    </div>
  )
}

Paginator.defaultProps = {
  currentPage: 1,
  isLastPage: false,
  onClickNext: () => {},
  onClickPrevious: () => {},
}

export default Paginator
