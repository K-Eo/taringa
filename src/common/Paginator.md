Basic Paginator

```js
<Paginator
  currentPage={2}
/>
```

First Page

```js
<Paginator
  currentPage={1}
/>
```

Last Page

```js
<Paginator
  currentPage={2}
  isLastPage
/>
```

One Page

```js
<Paginator
  currentPage={1}
  isLastPage
/>
```

Is Loading

```js
<Paginator
  isLoading
/>
```
