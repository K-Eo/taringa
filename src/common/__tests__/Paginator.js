import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Paginator from '../Paginator'

describe('Paginator', () => {
  it('renders default', () => {
    const tree = renderer.create(<Paginator currentPage={2} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders as first page', () => {
    const tree = renderer.create(<Paginator currentPage={1} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders as last page', () => {
    const tree = renderer
      .create(<Paginator currentPage={2} isLastPage />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders as one page', () => {
    const tree = renderer
      .create(<Paginator currentPage={1} isLastPage />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders as loading', () => {
    const tree = renderer.create(<Paginator isLoading />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('handles onClickPrevious', () => {
    const mock = jest.fn()
    const paginator = shallow(
      <Paginator currentPage={2} onClickPrevious={mock} />
    )

    paginator.find('button.Paginator-previous').simulate('click')

    expect(mock).toBeCalled()
  })

  it('handles onClickNext', () => {
    const mock = jest.fn()
    const paginator = shallow(<Paginator currentPage={2} onClickNext={mock} />)

    paginator.find('button.Paginator-next').simulate('click')

    expect(mock).toBeCalled()
  })
})
