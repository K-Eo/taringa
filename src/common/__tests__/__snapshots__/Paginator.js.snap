// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator renders as first page 1`] = `
<div
  className="Paginator"
>
   
  <button
    className="btn btn-primary btn-sm Paginator-next"
    onClick={[Function]}
  >
    Siguiente
  </button>
</div>
`;

exports[`Paginator renders as last page 1`] = `
<div
  className="Paginator"
>
  <button
    className="btn btn-primary btn-sm Paginator-previous"
    onClick={[Function]}
  >
    Anterior
  </button>
   
</div>
`;

exports[`Paginator renders as loading 1`] = `null`;

exports[`Paginator renders as one page 1`] = `
<div
  className="Paginator"
>
   
</div>
`;

exports[`Paginator renders default 1`] = `
<div
  className="Paginator"
>
  <button
    className="btn btn-primary btn-sm Paginator-previous"
    onClick={[Function]}
  >
    Anterior
  </button>
   
  <button
    className="btn btn-primary btn-sm Paginator-next"
    onClick={[Function]}
  >
    Siguiente
  </button>
</div>
`;
